lib/improver/tests/nowcasting/nowcasting/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,457 @@
+
+    """Test the __init__ method."""
+
+    def setUp(self):
+        """Set up cubes for testing."""
+        self.precip_cubes = set_up_precipitation_rate_cube()
+        self.oe_cubes = set_up_orographic_enhancement_cube()
+
+    def test_basic(self):
+        """Test that the plugin can be initialised as required."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.assertEqual(plugin.operation, "add")
+
+    def test_exception(self):
+        """Test that an exception is raised if the operation requested is
+        not a valid choice."""
+        msg = "Operation 'multiply' not supported for"
+        with self.assertRaisesRegex(ValueError, msg):
+            ApplyOrographicEnhancement("multiply")
+
+
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(ApplyOrographicEnhancement("add"))
+        msg = ('<ApplyOrographicEnhancement: operation: add>')
+        self.assertEqual(result, msg)
+
+
+class Test__extract_orographic_enhancement_cube(IrisTest):
+
+    """Test the _extract_orographic_enhancement method."""
+
+    def setUp(self):
+        """Set up cubes for testing."""
+        self.precip_cubes = set_up_precipitation_rate_cube()
+        self.oe_cube = set_up_orographic_enhancement_cube().concatenate_cube()
+        self.first_slice = self.oe_cube[:, 0, :, :]
+        self.second_slice = self.oe_cube[:, 1, :, :]
+
+    def test_basic(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube."""
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.first_slice)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_quarter_past(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at quarter past an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.25
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.first_slice.metadata)
+        self.assertEqual(result, self.first_slice)
+        self.assertEqual(result.coord("time"), self.first_slice.coord("time"))
+
+    def test_alternative_time_half_past(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at half past an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.51
```

Comment:
Why not 412227.5?  If the "nearest" behaviour is not stable / reproducible this is something we need to know.

Reply:
Changed to 412227.5 to show the behaviour.